include/libopencm3/usb/audio.h

```diff
@@ -88,6 +90,143 @@ struct usb_audio_header_descriptor_body {
 	uint8_t baInterfaceNr;
 } __attribute__((packed));
 
+/* Table 4-3: Input Terminal Descriptor */
+struct usb_audio_input_terminal_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bDescriptorSubtype;
+	uint8_t bTerminalID;
+	uint16_t wTerminalType;
+	uint8_t bAssocTerminal;
+	uint8_t bNrChannels;
+	uint16_t wChannelConfig;
+	uint8_t iChannelNames;
+	uint8_t iTerminal;
+} __attribute__((packed));
+
+/* Table 4-3: Output Terminal Descriptor */
+struct usb_audio_output_terminal_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bDescriptorSubtype;
+	uint8_t bTerminalID;
+	uint16_t wTerminalType;
+	uint8_t bAssocTerminal;
+	uint8_t bSourceID;
+	uint8_t iTerminal;
+} __attribute__((packed));
+
+/* Table 4-7: Feature Unit Descriptor (head) */
+struct usb_audio_feature_unit_descriptor_head {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bDescriptorSubtype;
+	uint8_t bUnitID;
+	uint8_t bSourceID;
+	uint8_t bControlSize;
+	uint16_t bmaControlMaster; /* device can assume 16-bit, given highest
+				    * defined bit in spec is bit #9.
+				    * (it is thus required bControlSize=2) */
```

Comment:
seems reasonable from the spec, and people could construct private ones with 
```
struct extended {
   struct usb_audio_feature_unit_descriptor_head head;
   struct private_extended_body body;
   struct usb_audio_feature_unit_descriptor_tail tail;
}
```
so we're not blocking anyone out either